ConnectionFactory.php

```diff
@@ -45,8 +45,21 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
-            $wrapperClass = null;
+        $allowOverrideUrl = $params['override_url'] ?? false;
+        unset($params['override_url']);
+
+        if (! $allowOverrideUrl) {
```

Comment:
I would add `&& isset($params['url'])`. there is no point forcing to configure `override_url` if `url` is not used at all.

Reply:
> I'm also fine with idea, but I think we should push for this to be a default. Perhaps we should make it deprecated to not set this option to true and in doctrine bundle 4 remove it and just make this behavior default. ...

we would still need to trigger the deprecation here regardless if `$params['url']` is set or not wouldn't we? based off of @ostrolucky idea that is.

Reply:
well, forcing them to add an option explicitly so that we can change the default does not make sense if they don't use url at all: they would not be affected by the change of default

Reply:
that makes sense and it's now implemented. thanks